src/plugins/vis_type_timelion/server/series_functions/es/index.js

```diff
@@ -96,20 +96,16 @@ export default new Datasource('es', {
       kibana: true,
       fit: 'nearest',
     });
-
-    const findResp = await tlConfig.savedObjectsClient.find({
-      type: 'index-pattern',
-      fields: ['title', 'fields'],
-      search: `"${config.index}"`,
-      search_fields: ['title'],
-    });
-    const indexPatternSavedObject = findResp.saved_objects.find((savedObject) => {
-      return savedObject.attributes.title === config.index;
+    const indexPatternsService = await tlConfig.getIndexPatternsService();
+    const searchIndexPatterns = await indexPatternsService.find(config.index);
+    const indexPatternSpec = searchIndexPatterns.find((indexPattern) => {
```

Comment:
@timroes not sure that I fully understand how it should be. Before we could not create two indexes with the same `title`. Now, `title` it's just a name and for mapping ES indexes we use `patternList` property. It means that I can do something like: 
![image](https://user-images.githubusercontent.com/20072247/106275798-7defce00-6247-11eb-8dbb-a83373f6546d.png)

Let's complicate the case a bit and make changes to the `f2` index. You can set a custom field or add a new scripted field. It does not matter. For my example I added a scripted field

![image](https://user-images.githubusercontent.com/20072247/106276140-269e2d80-6248-11eb-9c7a-8a97051db773.png)

now let's go back to that code. So: `searchIndexPatterns` - it's an array  which keeps 2 indexes (`f1, f2`). 'find' by array returns the first matched result. It's `f1`.   

But what if I need to work with `f2`? 

I really like that code and I think that it's a good enchantment for our users. But have no idea how `TSVB, Timelion and Vega` should work with that.